skeleton_scary_game.py

```diff
@@ -0,0 +1,329 @@
+            print("You have a gun. Ready to kill the demon, and get your daughter back?")
+            print("Now that you have a Deagle, you will only have to collect the bullets. Don't miss!")
+            # This are the items being removed from the bagpack
+            bagpack.remove("Gunpowder")
+            bagpack.remove("Magazine")
+            bagpack.remove("Pistol")
+
+            # The user gains a item since he had all the required parts
+            bagpack.append("Deagle")
+
+        else:
+            print("You don't have all the parts yet")
+
+    # This is where you can use the torch
+    def use_torch(self, bagpack, item_path):
+        if "Torch" in bagpack:
+            print(f"The item is on {item_path}. \n")
+            bagpack.remove("Torch")
+        else:
+            print("You do not have a torch\n")
+
+    # This is where you can use the pistol.
+    def use_pistol(self, bagpack):
+        if "Deagle" in bagpack and "Bullets" in bagpack:
+            print("You need to pick the path with the demon and you will kill him and get your daughter back\n")
+            bagpack.remove("Bullets")
+        else:
+            print("You do not have all the ")
+
+
+# This is the evil demon object, and it will be sent to the game_engine
+class Evil_demon(object):
+    """
+    This class is the demon class and really you just die if you encounter the demon.
+    """
+
+    # If  the user picks this path, the user dies
+    def death(self):
+        print("The demon ate you for dinner and let your child go back home parentless.")
+        exit(1)
+
+    def kill_demon(self):
+        print("You have killed the demon")
+        print("Daughter: DADDY you saved me, I knew you would come rescue me!!")
+        exit(0)
+
+    def missed_demon(self):
+        print("You shot but missed the demon. You will have to find another bullet and try again.")
+
+
+# This is the item object, and will be sent to the game engine as well.
+class Item(object):
+    """
+    This is the item class where the item list and the user bagpack is intialized and if
+    you walk the item path then you collect that item. Once the item has been used then it gets removed.
+    """
+
+    # These are the variables that are in this class. and will only be in this class.
+    def __init__(self, items_list, users_bagpack):
+        self.items_list = items_list
+        self.users_bagpack = users_bagpack
+
+    # If the user picks this path, the user collects the item and will be sent to his bagpack
+    def collect_item(self):
+        prize_item = random.choice(self.items_list)
+        print(f"You have survived and you have found => {prize_item}")
+        self.users_bagpack.append(prize_item)
+        print(f"You have {self.users_bagpack} \n")
+
+        # The user should not collect the same gun parts twice. It is a one and done thing
+        if prize_item == "Gunpowder" or "Pistol" or "Magazine":
```

Comment:
this logic isn't right here. yesterday you showed me an example of similar logic somewhere else.